jar/jar.go

```diff
@@ -59,10 +84,16 @@ type Report struct {
 	Version   string
 }
 
-// Parse traverses a JAR file, attempting to detect any usages of vulnerable
-// log4j versions.
-func Parse(r *zip.Reader) (*Report, error) {
-	var c checker
+// Parse traverses a JAR file, attempting to detect any usages of
+// vulnerable log4j versions.  If multiple ParseOptions are specified,
+// the last one takes precedence.
+func Parse(r *zip.Reader, o ...ParseOptions) (*Report, error) {
```

Comment:
We likely won't take PRs that break API compatibility. E.g. this breaks the following code

```
var parseFunc func(r *zip.Reader) = jar.Parse
```

Can you update to do something like:

```
type Parser struct {
    MaxDepth uint64
    MaxBytes uint64
}

func (p *Parser) Parse(r *zip.Reader) (*Report, error)
```

Reply:
Please switch to `Parse` since that's what the existing API looks like. I'd rather have

```
type Parser struct {}
func (p *Parser) Parse(...)
func Parse(...)
```

Than two names for the same API :)

```
type Checker struct {}
func (c *Checker) CheckJAR(...)
func Parse(...)
```

Reply:
Done.